packages/core/src/Dropdown/stories/CustomDropdown.tsx

```diff
@@ -0,0 +1,149 @@
+import { forwardRef, useState } from "react";
+import { css } from "@emotion/css";
+import {
+  HvIconButton,
+  HvInput,
+  HvSimpleGrid,
+  HvTreeItem,
+  HvTreeItemProps,
+  HvTreeView,
+  theme,
+} from "@hitachivantara/uikit-react-core";
+import { Doc, Folders, Plant } from "@hitachivantara/uikit-react-icons";
+
+import { HvBaseDropdown } from "../../BaseDropdown";
+import { HvPanel } from "../../Panel";
+
+type TreeData = { id: string; label: string; children?: TreeData[] };
+
+const treeDataObject = {
+  id: "user",
+  label: "User",
+  children: [
+    {
+      id: "Applications",
+      label: "Applications",
+      children: [
+        { id: "Code", label: "Code.app" },
+        { id: "Chrome", label: "Chrome.app" },
+        { id: "Firefox", label: "Firefox.app" },
+      ],
+    },
+    {
+      id: "Documents",
+      label: "Documents",
+      children: [{ id: "secret", label: "secret.txt" }],
+    },
+    {
+      id: "git",
+      label: "git",
+      children: [
+        {
+          id: "uikit-react",
+          label: "uikit-react",
+          children: [{ id: "uikit-pkg", label: "package.json" }],
+        },
+        {
+          id: "app-shell",
+          label: "app-shell",
+          children: [{ id: "as-pkg", label: "package.json" }],
+        },
+      ],
+    },
+  ],
+} satisfies TreeData;
+
+const SimpleTreeItem = forwardRef<HTMLLIElement, HvTreeItemProps>(
+  (props, ref) => {
+    const { children, nodeId, label, ...others } = props;
+    const Icon = children ? Folders : Doc;
+
+    return (
+      <HvTreeItem
+        ref={ref}
+        nodeId={nodeId}
+        label={
+          <div className={css({ display: "flex", alignItems: "center" })}>
+            <Icon />
+            <span style={{ flex: 1 }}>{label}</span>
+          </div>
+        }
+        {...others}
+      >
+        {children}
+      </HvTreeItem>
+    );
+  },
+);
+
+/** Render tree view items */
+const renderItem = ({ id, label, children }: TreeData) => (
+  <SimpleTreeItem key={id} nodeId={id} label={label}>
+    {children?.map(renderItem)}
+  </SimpleTreeItem>
+);
+
+export const CustomDropdown = () => {
+  const [results, setResults] = useState<TreeData | undefined>(treeDataObject);
+
+  const search = (data: TreeData, searchTerm: string): TreeData | undefined => {
+    let newChildren: TreeData[] = [];
+    if (data.label.includes(searchTerm)) {
```

Comment:
If I research "firefox" for example, the "Firefox" item is not shown because of the capitalised letter. We could add `toLowerCase`:

```suggestion
    if (data.label.toLowerCase().includes(searchTerm.toLowerCase())) {
```